examples/rust/examples/maker/maker.rs

```diff
@@ -42,23 +45,53 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let settings = Settings::load(&args[0]);
 
     if settings.node_endpoint.starts_with("http") {
-        let provider = Provider::<Http>::try_from(settings.node_endpoint.clone())?;
-        run(Arc::new(provider), settings).await;
+        let op = || async {
+            warn!("Re/starting maker");
+            run(Arc::new(Provider::<Http>::try_from(settings.node_endpoint.clone())?),
+                Settings::load(&args[0])).await
+        };
+
+        op.retry(
+            &ExponentialBuilder::default()
+                .with_jitter()
+                .with_max_times(10),
+        )
+            .await?;
     } else if settings.node_endpoint.starts_with("ws") {
         // Websockets (ws & wss)
-        let provider = Provider::<Ws>::new(Ws::connect(settings.node_endpoint.clone()).await?);
-        run(Arc::new(provider), settings).await;
+        let op = || async {
+            warn!("Re/starting maker");
+            run(Arc::new(Provider::<Ws>::new(Ws::connect(settings.node_endpoint.clone()).await?)),
+                Settings::load(&args[0])).await
+        };
+
+        op.retry(
+            &ExponentialBuilder::default()
+                .with_jitter()
+                .with_max_times(10),
+        )
+            .await?;
     } else {
         // IPC
-        let provider = Provider::connect_ipc(settings.node_endpoint.clone()).await?;
-        run(Arc::new(provider), settings).await;
+        let op = || async {
+            warn!("Re/starting maker");
+            run(Arc::new(Provider::connect_ipc(settings.node_endpoint.clone()).await?),
+                Settings::load(&args[0])).await
+        };
+
+        op.retry(
+            &ExponentialBuilder::default()
+                .with_jitter()
+                .with_max_times(10),
+        )
+            .await?;
     }
 
-    unreachable!("Run function has returned from a while true loop");
+    Ok(())
 }
 
 // Main execution function. This is not expected to return.
-async fn run<P: JsonRpcClient + 'static>(provider: Arc<Provider<P>>, settings: Settings) {
+async fn run<P: JsonRpcClient + 'static>(provider: Arc<Provider<P>>, settings: Settings) -> Result<(), anyhow::Error> {
```

Comment:
Given the signature change, wouldn't we also want to change the naked unwraps that are performed as well since we want to restart the entire process. There are also a couple of places within the `run` body where we clean exit while trying to initially connect to Valorem (i.e. establishment of the authenticated line), these should also be changed to return an error in an attempt to restart the process again.

If the retry was only required for connection to the node, then we shouldn't wrap the rety around the run as this implies that the maker should be retryable (which needs more changes to avoid panic scenarios)

Reply:
Confirmed within Discord this is still retry on panics/exits.